log-viewer/src/main/java/com/logviewer/data2/Log.java

```diff
@@ -228,6 +253,9 @@ private SeekableByteChannel getChannel() throws IOException {
                 if (error != null)
                     throw new IOException(error);
 
+                if (unpackArchive && Arrays.stream(GZIP_EXTENSIONS).anyMatch(it -> file.toString().toLowerCase().endsWith(it)))
```

Comment:
Let's use `TGZ.getPattern().matcher(file.toString()).matches()` to check if the file type is 'tgz'.